src/test/change_stream.rs

```diff
@@ -0,0 +1,711 @@
+use std::time::Duration;
+
+use futures::StreamExt;
+
+use crate::{
+    bson::doc,
+    change_stream::document::{ChangeStreamEventSource, OperationType, UpdateDescription},
+    options::{ChangeStreamOptions, FullDocumentType, UpdateOptions},
+    selection_criteria::{ReadPreference, SelectionCriteria},
+    test::{util::drop_collection, TestClient, LOCK},
+    Namespace,
+    RUNTIME,
+};
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn empty_change_stream() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    if client.server_version_lt(4, 0) || client.is_standalone() {
+        return;
+    }
+    let pipeline = vec![doc! { "$match": { "x": 3 } }];
+    let options = ChangeStreamOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .max_await_time(Duration::from_millis(5))
```

Comment:
5 ms is very short for a maxAwaitTimeMS; in practice, this will mean the change stream will send a getMore to the server as soon as it gets back an empty response, which is too frequent. This should be something higher (probably at least 500 ms).

Reply:
fixed